I'm a little confused by this; the error in the pixi ubuntu build log is:

```
2026-07-16T20:04:55.8913673Z ImportError: Failed to import test module: TestDraft
2026-07-16T20:04:55.8914390Z Traceback (most recent call last):
2026-07-16T20:04:55.8915340Z   File "/home/runner/work/FreeCAD/FreeCAD/.pixi/envs/default/lib/python3.11/unittest/loader.py", line 162, in loadTestsFromName
2026-07-16T20:04:55.8916269Z     module = __import__(module_name)
2026-07-16T20:04:55.8916671Z              ^^^^^^^^^^^^^^^^^^^^^^^
2026-07-16T20:04:55.8917440Z   File "/home/runner/work/FreeCAD/FreeCAD/.pixi/envs/default/Mod/Draft/TestDraft.py", line 100, in <module>
2026-07-16T20:04:55.8918332Z     from drafttests.test_import import DraftImport as DraftTest01
2026-07-16T20:04:55.8919385Z   File "/home/runner/work/FreeCAD/FreeCAD/.pixi/envs/default/Mod/Draft/drafttests/test_import.py", line 38, in <module>
2026-07-16T20:04:55.8920254Z     from drafttests import test_base
2026-07-16T20:04:55.8921133Z   File "/home/runner/work/FreeCAD/FreeCAD/.pixi/envs/default/Mod/Draft/drafttests/test_base.py", line 31, in <module>
2026-07-16T20:04:55.8921966Z     from draftutils.todo import ToDo
2026-07-16T20:04:55.8922829Z   File "/home/runner/work/FreeCAD/FreeCAD/.pixi/envs/default/Mod/Draft/draftutils/todo.py", line 45, in <module>
2026-07-16T20:04:55.8923640Z     import FreeCADGui as Gui
2026-07-16T20:04:55.8924537Z ImportError: libEGL.so.1: cannot open shared object file: No such file or directory
```

libEGL is installed and available in pixi environments

```
find .pixi -name "*EGL.so*"
.pixi/envs/default/lib/libEGL.so
.pixi/envs/default/lib/libEGL.so.1
.pixi/envs/default/lib/libEGL.so.1.1.0
```

Could it be a RPATH problem coming from whatever library is loading EGL, not linking to it correctly?

While your solution is fine, we could also restore the old behavior of using the system libEGL by installing it using `apt` on the runner.

--- a/cMake/FreeCAD_Helpers/SetupCoin3D.cmake
+++ b/cMake/FreeCAD_Helpers/SetupCoin3D.cmake
@@ -126,6 +126,12 @@ macro(SetupBundledCoinPivy)
         "$<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/src/3rdParty/coin/include>"
         "$<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/src/3rdParty/coin/include>"
     )
+    if (UNIX AND NOT APPLE)
+        # Coin directly loads sibling libraries such as libEGL.  The RUNPATH of
+        # a library that loads Coin is not used to resolve Coin's dependencies.
+        # Keep the bundled library relocatable in both Pixi and AppImage installs.
+        set_property(TARGET Coin PROPERTY INSTALL_RPATH "$ORIGIN")
+    endif ()
     install(TARGETS Coin
         RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
         LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} NAMELINK_SKIP

